examples/codenames/Codenames_RL.ipynb

```diff
@@ -0,0 +1,12542 @@
+      "execution_count": null,
+      "metadata": {
+        "id": "xA5ADGKPjakD"
+      },
+      "outputs": [],
+      "source": [
+        "%%capture\n",
+        "!uv pip install openpipe-art openpipe --prerelease allow --no-cache-dir"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C5n8N1uSjakD"
+      },
+      "source": [
+        "# Agentic Environment\n",
+        "<a name=\"Environment\"></a>\n",
+        "\n",
+        "ART allows your agent to learn by interacting with its environment. In this example, we'll create an environment in which the agent can play Codenames.\n",
+        "\n",
+        "Feel free to read as much or as little of this section's code as you'd like. The important thing to understand is that we're defining the rules of this agent's environment. In many cases, this will already be defined by the task you're trying to solve, but if you need to define a custom environment, this is how you do it."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "K9h-g67egOpy",
+        "outputId": "f6489f2d-a795-4dc5-82ba-24a0cda6dbea"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+            "100  2481  100  2481    0     0   6941      0 --:--:-- --:--:-- --:--:--  6949\n",
+            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+            "100 4749k  100 4749k    0     0  5151k      0 --:--:-- --:--:-- --:--:-- 5150k\n"
+          ]
+        }
+      ],
+      "source": [
+        "# Download the 400 original codenames words and a dictionary to ensure that the clue is not outside of this dictionary\n",
+        "!curl -o codenames_words.txt https://raw.githubusercontent.com/saum7800/codenames-public-data/refs/heads/main/codenames_words.txt\n",
+        "!curl -o dictionary.txt https://raw.githubusercontent.com/saum7800/codenames-public-data/refs/heads/main/dictionary.txt"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "we0PhGpVoGmR",
+        "outputId": "fc39772e-7f84-4dc4-8915-ff1125636b2f"
+      },
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "/workspace/codenames-rl/.venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
```

Comment:
We should clear any outputs whose content isn't relevant to a user who wants to understand the notebook without running it.